### DISCUSSION TITLE
ERR_EMPTY_RESPONSE when trying to access Firefly

### DISCUSSION THREAD

**Discussion by Xico26** [firefly-iii/firefly-iii#9258] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hello. 

I have installed Firefly III via Portainer, using the steps in the documentation. However, when I access https://raspberrypi.local:5005 (I have changed the port due to some conflicts), I get an empty response (ERR_EMPTY_RESPONSE).
I have also changed the DB port to 3308, for the same reason, and added MYSQL_TCP_PORT and MYSQL_UNIX_PORT as env variables, like suggested by some research I did.
I have also removed the MYSQL_RANDOM_ROOT_PASSWORD env variable, and added a MARIADB_ROOT_PASSWORD. 

Checking the logs, I have noticed the db logs have the following: `3 [Warning] Aborted connection 3 to db: 'unconnected' user: 'unauthenticated' host: '172.18.0.4' (This connection closed normally without authentication)`, which led me to believe there was something wrong with the database, and, as I have seen recommended in https://github.com/orgs/firefly-iii/discussions/8984, I have tried to log in to the database as root, which did not work (`ERROR 1045 (28000): Access denied for user 'root'@'172.18.0.1' (using password: YES)`).

Using phpmyadmin, the users table is empty, which I am not sure if it is the intended behaviour.

I have reinstalled Firefly multiple times, either through Portainer or regular Docker, but I never get a response from the web page.

Here is the docker-composef file:

```docker-compose
#
# The Firefly III Data Importer will ask you for the Firefly III URL and a "Client ID".
# You can generate the Client ID at http://localhost/profile (after registering)
# The Firefly III URL is: http://app:8080
#
# Other URL's will give 500 | Server Error
#

services:
  app:
    image: fireflyiii/core:latest
    hostname: app
    container_name: firefly_iii_core
    networks:
      - firefly_iii
    restart: always
    volumes:
      - firefly_iii_upload:/var/www/html/storage/upload
    env_file: stack.env
    ports:
      - '5005:5005'
    depends_on:
      - db
  db:
    image: mariadb:lts
    hostname: db
    container_name: firefly_iii_db
    networks:
      - firefly_iii
    restart: always
    env_file: stack.env
    volumes:
      - firefly_iii_db:/var/lib/mysql
    ports:
      - 3308:3308

  importer:
    image: fireflyiii/data-importer:latest
    hostname: importer
    restart: always
    container_name: firefly_iii_importer
    networks:
      - firefly_iii
    ports:
      - '5007:5007'
    depends_on:
      - app
    env_file: stack.env

  cron:
    #
    # To make this work, set STATIC_CRON_TOKEN in your .env file or as an environment variable and replace REPLACEME below
    # The STATIC_CRON_TOKEN must be *exactly* 32 characters long
    #
    image: alpine
    container_name: firefly_iii_cron
    restart: always
    command: sh -c "echo \"0 3 * * * wget -qO- http://app:8080/api/v1/cron/REPLACEME\" | crontab - && crond -f -L /dev/stdout"
    networks:
      - firefly_iii
volumes:
   firefly_iii_upload:
   firefly_iii_db:

networks:
  firefly_iii:
    driver: bridge

```

### Debug information

N/A

### Expected behaviour

I expected the Firefly UI to load.

### Steps to reproduce

1. Create a stack using the docker compose file I used.
2. Set up the env variables mentioned in the documentation and the changes I made.
3. Try to access the page

**Comment by JC5** [comment]
Hi, thanks for opening an issue. 

Why is this a bug, and not a configuration issue?

**Comment by Xico26** [comment]
Hi. I understand this could be some issue on my part, with my configuration (but I don't know what the issue is). 
However, I have done multiple installs, using both Portainer and Docker, always following the documentation, and I have always had this issue. In the first install, I did exactly what the docs said, without any modifications, and still got the error. That's what made me try Portainer and changing ports, but it clearly didn't fix the issue.

**Comment by JC5** [comment]
The 'unauthenticated' error at least is normal, that's nothing to worry about.

Is there a reverse proxy in front of Firefly III?

**Comment by Xico26** [comment]
Not that I'm aware of, no.
I managed to fix the issue by changing the VANITY_URL variable to my Pi's address (http://raspberrypi.local:8080), instead of just http://localhost, like mentioned in the documentation.
